src/abstractdataframe/join.jl

```diff
@@ -303,9 +304,12 @@ function Base.join(df1::AbstractDataFrame,
                    Symbol(indicator_cols[i]) == indicator)
                  indicator_cols[i] *= 'X'
             end
-         end
-         df1 = hcat(df1, DataFrame(Dict(indicator_cols[1] => trues(nrow(df1)))))
-         df2 = hcat(df2, DataFrame(Dict(indicator_cols[2] => trues(nrow(df2)))))
+        end
+        # TODO: clean up when df[col] syntax is deprecated
```

Comment:
This seems redundant with the deprecation that we will print to catch these.

Below, I'd find it more natural to use a shallow `copy`.

Reply:
OK - I have changed to `copy`.
I prefer to retain these `TODO` clauses exactly because when we decide what we do with https://github.com/JuliaData/DataFrames.jl/issues/1753 it will be easier to find all parts of the code that need to be looked at (and this should happen before the next release anyway).

Reply:
Can't you just post the list of places you spotted at #1753?

Reply:
I wil just remove it (it is me who has to track it down later anyway - I will manage 😄)